task/spdm/src/main.rs

```diff
@@ -0,0 +1,183 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#![no_std]
+#![no_main]
+
+// use idol_runtime::{ClientError, Leased, LenLimit, RequestError, R, W};
+use idol_runtime::{ClientError, Leased, RequestError, R, W};
+use task_spdm_api::SpdmError;
+use userlib::*;
+
+use ringbuf::*;
+use spdm::{
+    config::NUM_SLOTS,
+    crypto::{FakeSigner, FilledSlot},
+    responder::AllStates,
+};
+
+#[derive(Copy, Clone, PartialEq, Debug)]
+enum State {
+    Error,
+    Version,
+    Capabilities,
+    Algorithms,
+    IdAuth,
+    Challenge,
+}
+
+impl From<&AllStates> for State {
+    fn from(state: &AllStates) -> Self {
+        match state {
+            AllStates::Error => State::Error,
+            AllStates::Version(_) => State::Version,
+            AllStates::Capabilities(_) => State::Capabilities,
+            AllStates::Algorithms(_) => State::Algorithms,
+            AllStates::IdAuth(_) => State::IdAuth,
+            AllStates::Challenge(_) => State::Challenge,
+        }
+    }
+}
+
+/// Record the types and sizes of the messages sent and received by this server
+#[derive(Copy, Clone, PartialEq, Debug)]
+enum LogMsg {
+    // Static initializer
+    Init,
+    _State(State),
+}
+ringbuf!(LogMsg, 16, LogMsg::Init);
+
+const MAX_SPDM_MSG_SIZE: usize = 256;
+
+#[export_name = "main"]
+fn main() -> ! {
+    const EMPTY_SLOT: Option<FilledSlot<'_, FakeSigner>> = None;
+    let _slots = [EMPTY_SLOT; NUM_SLOTS];
+    // let responder = spdm::Responder::new(slots);
+    // ringbuf_entry!(LogMsg::State(responder.state().into()));
+
+    let mut buffer = [0; idl::INCOMING_SIZE];
+    let msg = [0; MAX_SPDM_MSG_SIZE];
+    let mut server = ServerImpl {
+        // _responder: responder,
+        valid: 0,
+        message: msg,
+    };
+
+    loop {
+        idol_runtime::dispatch(&mut buffer, &mut server);
+    }
+}
+
+struct ServerImpl {
+    // _responder: spdm::Responder<'a, FakeSigner>,
+    valid: usize,
+    message: [u8; MAX_SPDM_MSG_SIZE],
+}
+
+impl idl::InOrderSpdmImpl for ServerImpl {
+    /// A client sends a message for SPDM processing.
+    fn send(
```

Comment:
The `send` and `recv` terminology from a server PoV is already confusing me. It makes sense from a client/humility POV where we use humility as a "step based" debugger, but I don't know why a SPDM requester or responder would need to be told to `send`. A requester does need to be told when to send `GET_MEASUREMENTS` perhaps, but that could also be encoded in the SPDM requester code. There will also be some proxy behavior where the SP essentially just forwards signing requests to the RoT. 

I'm thinking we'll likely end up just wanting semantic operations in humility and we may or may not want to map them 1:1 with IDL.

In short, I know I pushed a bit for send/recv, but maybe we should remove them and just use `exchange` for now while we build out the SPDM responder code only here. We can use humility as the requester and once we have a flow worked out we can go back and change the IDL...

I think